packages/devtools_app/lib/src/debugger/debugger_controller.dart

```diff
@@ -585,33 +592,55 @@ class DebuggerController extends DisposableController
         [pauseEvent.topFrame],
         reportedException: pauseEvent?.exception,
       );
-      _stackFramesWithLocation.value = [
-        await _createStackFrameWithLocation(tempFrames.first),
-      ];
+      _populateFrameInfo(
+          [await _createStackFrameWithLocation(tempFrames.first)],
+          truncated: true);
       _log.log('created first frame');
-      selectStackFrame(_stackFramesWithLocation.value.first);
     }
 
-    // Then, issue an asynchronous request to populate the frame information.
-    _log.log('getStack()');
-    final stack = await _service.getStack(isolateRef.id);
+    // We populate the first 10 frames to match the behavior in VS Code.
+    _getStackOperation =
+        CancelableOperation.fromFuture(_getStackInfo(limit: 10));
+    final stackInfo = await _getStackOperation.value;
+    _populateFrameInfo(stackInfo.frames, truncated: stackInfo.truncated);
+  }
+
+  Future<_StackInfo> _getStackInfo({int limit}) async {
+    _log.log('getStack() with limit: $limit');
+    final stack = await _service.getStack(isolateRef.id, limit: limit);
     _log.log('getStack() completed (frames: ${stack.frames.length})');
+
     final frames = _framesForCallStack(
       stack.frames,
       asyncCausalFrames: stack.asyncCausalFrames,
-      reportedException: pauseEvent?.exception,
+      reportedException: _lastEvent?.exception,
     );
 
-    _stackFramesWithLocation.value =
-        await Future.wait(frames.map(_createStackFrameWithLocation));
+    return _StackInfo(
+        await Future.wait(frames.map(_createStackFrameWithLocation)),
+        stack.truncated);
+  }
+
+  void _populateFrameInfo(List<StackFrameAndSourcePosition> frames,
+      {bool truncated}) {
+    truncated ??= false;
```

Comment:
instead of doing this, just move the default value into the method signature

Reply:
Until NNBD that's actually not a good pattern. Users can still explicitly pass in null which will cause issues.